src/apps/lwaftr/ipv6_apps.lua

```diff
@@ -147,3 +152,61 @@ function Fragmenter:push ()
       end
    end
 end
+
+function ICMPEcho:new(conf)
+   local addresses = {}
+   if conf.address then
+      addresses[ffi.string(conf.address, 16)] = true
+   end
+   if conf.addresses then
+      for _, v in ipairs(conf.addresses) do
+         addresses[ffi.string(v, 16)] = true
+      end
+   end
+   return setmetatable({addresses = addresses}, {__index = ICMPEcho})
```

Comment:
`address` is a 32-bit cdata object, correct?  If that's the case it's possible to convert the 32-bit cdata to a Lua number and use it as an index, probably is cheaper to run `tonumber()` than `ffi.tostring()`.

OTOH, there's a `set` function in `dump.lua`, probably not the best place. Since you need to create a set, maybe you can move this function to `lwutils.lua` and reuse it. I think using the `set` function makes the code easier to read.